### DISCUSSION TITLE
Add metadata information about Required Actions to the ftl template

### DISCUSSION THREAD

**Discussion by jairsjunior** [keycloak/keycloak#12464] [Ideas]
When we are handling one Required Action sometimes more than one template is being used (like `info.ftl` file). And from the template perspective, it's not possible to know who is loading this FTL and act accordingly with is needed. Another thing is to have proper analytics being collected from the Front end side, using this metadata.

The proposal here is to add some metadata about the Required Action execution to the object shared with the FTL file like:

- Required action name that is being executed;
- Previous action name that was executed (if exists on the session);
- Next action name that will be executed (if exists on the session);

Do you think that this improvement will help more Keycloak developers/users?

**Comment by bicisteadm** [comment]
I currently ran into the same problem, where in info.ftl I need to know what action it is. Is there any solution?

**Comment by DaBonse** [comment]
Same here, we also need to know what action ist is in info.ftl template.
